From e2dc16cb2004a87994641eed461ca87f620d782e Mon Sep 17 00:00:00 2001
From: Chris Morgan <macromorgan@hotmail.com>
Date: Wed, 30 Oct 2024 17:47:39 -0500
Subject: [PATCH 02/10] arm64: dts: rockchip: Add spk_amp regulator for
 Anbernic RGxx3 Devices

Define the VCC supply for the speaker amp on the Anbernic RGxx3 series
devices which use an external amplifier circuit (all do except the V
and VS series). This supply value was discovered by slightly raising
and lowering the voltages of various regulators from within the
bootloader and measuring for corresponding changes with a multimeter.

Signed-off-by: Chris Morgan <macromorgan@hotmail.com>
---
 arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg-arc.dtsi | 1 +
 arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353p.dts  | 1 +
 arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353ps.dts | 1 +
 arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg503.dts   | 1 +
 4 files changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg-arc.dtsi b/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg-arc.dtsi
index a4a60e4a53d4..012490092420 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg-arc.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg-arc.dtsi
@@ -72,6 +72,7 @@ spk_amp: audio-amplifier {
 		pinctrl-0 = <&spk_amp_enable_h>;
 		pinctrl-names = "default";
 		sound-name-prefix = "Speaker Amp";
+		VCC-supply = <&vcc_3v3>;
 	};
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353p.dts b/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353p.dts
index a73cf30801ec..ec6b42f8383d 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353p.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353p.dts
@@ -74,6 +74,7 @@ spk_amp: audio-amplifier {
 		pinctrl-0 = <&spk_amp_enable_h>;
 		pinctrl-names = "default";
 		sound-name-prefix = "Speaker Amp";
+		VCC-supply = <&vcc_3v3>;
 	};
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353ps.dts b/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353ps.dts
index ca5284e4807d..a59eff0c78d6 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353ps.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg353ps.dts
@@ -73,6 +73,7 @@ spk_amp: audio-amplifier {
 		pinctrl-0 = <&spk_amp_enable_h>;
 		pinctrl-names = "default";
 		sound-name-prefix = "Speaker Amp";
+		VCC-supply = <&vcc_3v3>;
 	};
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg503.dts b/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg503.dts
index 74cf313e0635..59e0448cb858 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg503.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rg503.dts
@@ -163,6 +163,7 @@ spk_amp: audio-amplifier {
 		pinctrl-0 = <&spk_amp_enable_h>;
 		pinctrl-names = "default";
 		sound-name-prefix = "Speaker Amp";
+		VCC-supply = <&vcc_3v3>;
 	};
 };
 
-- 
2.43.0

